src/pages/VenuePage/VenuePage.tsx

```diff
@@ -93,57 +94,49 @@ const VenuePage: React.FC = () => {
 
   // NOTE: User location updates
 
-  const updateUserLocationToCurrentVenue = useCallback(() => {
-    if (!userId || !venueName) return;
-
-    trackLocationEntered({ userId, locationName: venueName });
-  }, [userId, venueName]);
-
   useInterval(() => {
-    updateUserLocationToCurrentVenue();
+    if (!userId || !profile?.lastSeenIn) return;
+
+    updateCurrentLocationData({
+      userId,
+      profileLocationData: profile.lastSeenIn,
+    });
   }, LOC_UPDATE_FREQ_MS);
 
   useEffect(() => {
-    updateUserLocationToCurrentVenue();
-  }, [updateUserLocationToCurrentVenue]);
-
-  // useEffect(() => {
-  //   if (!userId || !venueName) return;
-  //   updateUserLocationToCurrentVenue();
-
-  // NOTE: A suggestion on how to avoid location cleaning, when two tabs were opened and one of them was closed
+    if (!userId || !venueName) return;
 
-  // document.addEventListener("visibilitychange", () => {
-  //   if (document.visibilityState === "visible") {
-  //     updateUserLocationToCurrentVenue();
-  //   }
-  // });
-  // }, [userId, venueName, updateUserLocationToCurrentVenue]);
+    setLocationData({ userId, locationName: venueName });
+  }, [userId, venueName]);
 
   useEffect(() => {
     if (!userId) return;
 
+    const onBeforeUnloadHandler = () => clearLocationData(userId);
+
     // NOTE: Clear user location on page close
-    window.addEventListener("beforeunload", () =>
-      updateLocationData(userId, {})
-    );
+    window.addEventListener("beforeunload", onBeforeUnloadHandler);
+
+    return () =>
+      window.removeEventListener("beforeunload", onBeforeUnloadHandler);
   }, [userId]);
 
   useEffect(() => {
     if (
-      profile?.enteredVenueIds &&
-      venueId &&
-      profile?.enteredVenueIds.includes(venueId)
-    )
+      !venueId ||
+      !userId ||
+      !profile ||
+      profile?.enteredVenueIds?.includes(venueId)
+    ) {
       return;
-    if (!venueId || !user || !profile) return;
+    }
 
-    updateProfileEnteredVenueIds(profile?.enteredVenueIds, user?.uid, venueId);
-  }, [profile, user, venueId]);
+    updateProfileEnteredVenueIds(profile?.enteredVenueIds, userId, venueId);
+  }, [profile, userId, venueId]);
```

Comment:
It would be better for this function to not have to pass in `enteredVenueIds` directly, but instead to use firebase's `arrayUnion` helper

- https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array

It would also be good to use a props object with named keys since there are multiple parameters.

Reply:
[OOS] https://github.com/sparkletown/internal-sparkle-issues/issues/302